packages/cli/src/commands/add/changeTypes.ts

```diff
@@ -0,0 +1,236 @@
+import { log } from "@changesets/logger";
+import {
+  ChangeType,
+  ChangesetWithConfirmed,
+  EmptyString,
+  Release,
+  Config,
+  VersionType
+} from "@changesets/types";
+import chalk from "chalk";
+
+import * as cli from "../../utils/cli-utilities";
+import changeTypeList from "./changeTypeList.json";
+
+const { bold, cyan } = chalk;
+
+type ChangeTypeOption = { title: string; text: string };
+const allChangeTypes: ChangeTypeOption[] = changeTypeList;
+
+type PreviousAnswers = { [key in string]: string };
+export class ChangesetsWithChangeTypes {
+  private config: Config;
+  private isWithChangeTypes: boolean = false;
+  private releases: Release[] = [];
+  private chosenChangeTypeList: ChangeTypeOption[] = [];
+  private changesetList: ChangesetWithConfirmed[] = [];
+
+  constructor(config: Config) {
+    this.config = config;
+  }
+
+  async setChangeTypeList() {
+    if (!this.config.shouldAskForChangeTypes) return;
+    const choosenTitleList = await getChangeTypeList();
+
+    this.chosenChangeTypeList = choosenTitleList.map(title => ({
+      title,
+      text: allChangeTypes.find(cht => cht.title === title)!.text
+    }));
+
+    this.isWithChangeTypes = choosenTitleList?.length > 0;
+  }
+
+  async setReleases(newReleases: Release[]) {
+    if (!this.isWithChangeTypes) return;
+    this.releases = newReleases;
+  }
+
+  async setChangesetList() {
+    if (!this.isWithChangeTypes) return;
```

Comment:
should we use typescript decorators for each guard `if (!this.isWithChangeTypes) return;` ?